apple/internal/apple_framework_import.bzl

```diff
@@ -314,12 +344,95 @@ def _debug_info_binaries(
 
     return all_binaries_dict.values()
 
-def _apple_dynamic_framework_import_impl(ctx):
-    """Implementation for the apple_dynamic_framework_import rule."""
+def _get_current_library_identifier(
+        *,
+        current_platform,
+        xcframework_path,
+        xcframework_imports):
+    library_identifiers = sets.make()
+
+    xcframework_path_length = len(xcframework_path)
+    for f in xcframework_imports:
+        inner_path = f.path[xcframework_path_length + 1:]
+        for i in range(len(inner_path)):
+            if inner_path[i] == "/":
+                identifier = inner_path[:i]
+                sets.insert(library_identifiers, identifier)
+                break
+
+    platform_type = str(current_platform.platform_type).lower()
+    is_device = current_platform.is_device
+    for id in sets.to_list(library_identifiers):
+        if not id.startswith(platform_type):
+            continue
+        if is_device and (id.endswith("-simulator") or id.endswith("-maccatalyst")):
+            continue
+        elif id.endswith("-simulator"):
+            return id
+
+    # TODO: Handle maccatalyst variant
+    return None
+
+def _get_xcframework_imports(ctx):
+    xcframework_path = ""
+    for f in ctx.files.xcframework_imports:
+        if ".xcframework" in f.path:
+            xcframework_path = f.path.split(".xcframework")[0] + ".xcframework"
+            break
+    if not xcframework_path.endswith(".xcframework"):
```

Comment:
I think we should vaildated there is only 1 xcframework path from this calculation. I doubt many folks use it but technically `framework_imports` can be used today to include multiple frameworks with different names. Having a `fail()` for this would just give us some guardrails for that case

Reply:
Added a check in the `_grouped_xcframework_files` function.